docs/allure-reporting.md

```diff
@@ -0,0 +1,46 @@
+# Generating and publishing Allure reports to share executable test results
```

Comment:
since you're adding docs - would you go ahead and re-organize the `doc/` folder, so most documents are in one of 4 subdirectories:
```
doc/tutorial/
doc/howto/
doc/explain/
doc/reference/
```

I like the way this bLog that Andrew posted before thinks about documentation: 
     https://www.divio.com/blog/documentation/
Maybe link to that in `doc/README.md` with the description of what each type of documentation is about:

* Tutorials
    - is learning-oriented
    - allows the newcomer to get started
    - is a lesson

* How-to guides
    - is goal-oriented
    - shows how to solve a specific problem
    - is a series of steps


* Explanation
    - is understanding-oriented
    - explains
    - provides background and context

* Reference
    - is information-oriented
    - describes the machinery
    - is accurate and complete

Reply:
I've created the folders and moved some `.md` files around. Please lemme know if you agree.